nbs/03_matcher.ipynb

```diff
@@ -10,7 +10,7 @@
   },
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #27.        if collections_pattern is not None:
</pre><p>this body is getting hard to read, especially lines 38-39.</p><p> </p><p>Try to see if you can simplify it.</p><p> </p><p>i would try to:</p><ul><li>make a helper function that output potentially non unique matches</li><li>then make the main find matches function yield from it and pass it to _remove duplicates</li><li>make the helper function a nested for loop over matches and then possible collection matches and then checks the condition and yields</li><li>do we even allow a condition on anonymous nodes? how do they reference the anonymous nodes? seems weird.</li><li>if we still need the anonymous nodes for the condition, why not make the non filtered match first, check the condition and only if its tru make the filtered one? why instantiate them both in a list and carry that list of pairs forward?</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>